src/main/java/com/sproutsocial/nsq/Batcher.java

```diff
@@ -87,6 +93,26 @@ public void run() {
     }
 
     void sendBatch() {
+        if (asyncPublish) {
+            sendBatchAsync();
+        } else {
+            sendBatchSync();
+        }
+    }
+
+    /**
+     * If you want asynch batching to never block message publishing, make sure you set aggressive socket timeouts.
+     * Otherwise you'll get stalls when there are batches outstanding.
+     */
+    private void sendBatchAsync() {
+        executor.submit(new Runnable() {
```

Comment:
@robseed I think there's a race condition here, where the scheduled / delayed call might need to be cancelled before it's safe to execute this, otherwise, we'd get double publishing.

Reply:
Scratch that, given that `sendDelayedBatch` will reschedule the next timeout check in a synchronized block, I think we're ok.

Reply:
:wave: Will this run into trouble when`publisher.publish()` gets slow and your application is producing messages really quickly? I think the task queue in your [executor](https://github.com/sproutsocial/nsq-j/blob/master/src/main/java/com/sproutsocial/nsq/Publisher.java#L176) is unbounded. I wonder if a bounded, drop-when-full queue might be a better choice? Or some other latching mechanism to avoid submitting successive tasks for the same batch?

Reply:
@ccstolley Heyo! My solution was to set aggressive timeouts, but it's probably a good idea to bound the executor as well to something very small. Good call.